libcaf_core/caf/actor_cast.test.cpp

```diff
@@ -0,0 +1,258 @@
+// This file is part of CAF, the C++ Actor Framework. See the file LICENSE in
+// the main distribution directory for license terms and copyright or visit
+// https://github.com/actor-framework/actor-framework/blob/main/LICENSE.
+
+#include "caf/actor_cast.hpp"
+
+#include "caf/test/fixture/deterministic.hpp"
+#include "caf/test/test.hpp"
+
+#include "caf/event_based_actor.hpp"
+#include "caf/typed_event_based_actor.hpp"
+
+using namespace caf;
+
+namespace {
+
+struct dummy_trait {
+  using signatures = type_list<result<int>(int)>;
+};
+
+using dummy_actor = typed_actor<dummy_trait>;
+
+behavior dummy_impl() {
+  return {
+    [](int x) { return x; },
+  };
+}
+
+dummy_actor::behavior_type typed_dummy_impl() {
+  return {
+    [](int x) { return x; },
+  };
+}
+
+struct access_tag {};
+
+} // namespace
+
+namespace caf {
+
+// Abuse the fact that `actor_cast_access` has friend access to the handles.
+template <>
+class actor_cast_access<access_tag, access_tag, -1> {
+public:
+  static auto& unbox(actor& x) {
+    return x.ptr_;
+  }
+
+  template <class... Ts>
+  static auto& unbox(typed_actor<Ts...>& x) {
+    return x.ptr_;
+  }
+
+  static auto& unbox(actor_addr& x) {
+    return x.ptr_;
+  }
+};
+
+using access_t = actor_cast_access<access_tag, access_tag, -1>;
+
+} // namespace caf
+
+WITH_FIXTURE(test::fixture::deterministic) {
+
+TEST("actor_cast converts a strong pointer to a weak  pointer") {
```

Comment:
Small style nitpick.
```suggestion
TEST("actor_cast converts a strong pointer to a weak pointer") {
```